pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsNotNullTransformFunction.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.pinot.core.operator.transform.function;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.reader.NullValueVectorReader;
+
+
+public class IsNotNullTransformFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "IS_NOT_NULL";
+
+  private TransformFunction _leftTransformFunction;
+  private int[] _results;
+  private Map<String, DataSource> _dataSourceMap = new HashMap<>();
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    Preconditions.checkArgument(arguments.size() == 1,
+        "Exact 1 argument is required for IS_NOT_NULL operator function");
+    _leftTransformFunction = arguments.get(0);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException(
+          "Only column names are supported in IS_NOT_NULL. Support for functions is planned for future release");
+    }
+    _dataSourceMap = dataSourceMap;
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    if (_results == null || _results.length < length) {
+      _results = new int[length];
+    }
+
+    int[] docIds = projectionBlock.getDocIds();
+    String columnName = ((IdentifierTransformFunction) _leftTransformFunction).getColumnName();
+    NullValueVectorReader nullValueVector = _dataSourceMap.get(columnName).getNullValueVector();
+
+    if (nullValueVector != null) {
+      for (int idx = 0; idx < length; idx++) {
+        int docId = docIds[idx];
+        if (nullValueVector.isNull(docId)) {
+          _results[idx] = 0;
+        } else {
+          _results[idx] = 1;
+        }
+      }
```

Comment:
It would be much more efficient to make the null value vector iterable in a range and set the value to 0 whenever the null value vector has a value.

Reply:
Same below - the reason is the complexity is O(n * log (m)) where n is the number of rows and m is the number of null values, but it could be O(n) (fill the array with ones then iterate over the null value bitmap).

Reply:
That will require extra space complexity of O(n) to store docId to index lookup. Or is there any other way?

Reply:
No it won't - the null value vector is just a bitmap. If docId x is null, bit x is set in the null value vector.

Reply:
Try something like this:

```java
if (nullValueVector != null) {
      PeekableIntIterator it = nullValueVector.getNullBitmap().getIntIterator();
      int pos = 0;
      while (it.hasNext() & pos < length) {
        it.advanceIfNeeded(docIds[pos]);
        pos = Arrays.binarySearch(docIds, pos, length, it.next());
        if (pos >= 0) {
          _results[pos] = 0;
          pos++;
        } else {
          pos = -pos - 1;
        }
      }
    }
```

Since blocks are iterated in ascending docId order, it would be better to store the iterator in the TransformFunction which would avoid mapping and advancing the bitmap once per block (currently this code maps the bitmap once per row).

Reply:
Also, it should be easy to extend `BenchmarkQueries` to test this out